src/extension.ts

```diff
@@ -109,6 +112,56 @@ export function activate(context: vscode.ExtensionContext): void {
 		}
 	});
 
+	// Initializes worker for diffing
+	let worker: Worker;
+	const workerFilePath = vscode.Uri.joinPath(
+		context.extensionUri,
+		"dist",
+		"diffWorker.js",
+	).toString();
+
+	try {
+		worker = new Worker(workerFilePath);
+	} catch {
+		// eslint-disable-next-line @typescript-eslint/no-var-requires
+		const { Worker } = require("worker_threads") as typeof import("worker_threads");
+		const nodeWorker = new Worker(new URL(workerFilePath));
+		// Web and node js have different worker interfaces, so we share a function
+		// to initialize both workers the same way.
+		const ref = nodeWorker.addListener;
+		(nodeWorker as any).addEventListener = ref;
+		worker = nodeWorker as any;
+	}
+
+	const workerMessageHandler = new MessageHandler<ToDiffWorkerMessage, FromDiffWorkerMessage>(
+		// Always return undefined as the diff worker
+		// does not request anything from extension host
+		async () => undefined,
+		message => worker.postMessage(message),
+	);
+
+	worker.addEventListener("message", e =>
+		// e.data is used in web worker and e is used in node js worker
+		e.data ? workerMessageHandler.handleMessage(e.data) : workerMessageHandler.handleMessage(e as any),
+	);
```

Comment:
I have done both, but I still kept the dispose to be done when the extension is deactivated since I am not entirely sure when to dispose it. Should I keep it like this?